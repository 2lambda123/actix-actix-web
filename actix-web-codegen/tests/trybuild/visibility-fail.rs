use actix_web::App;

mod config {
    use actix_web_codegen::*;
    use actix_web::{Responder, HttpResponse};
    
    #[get("/config")]
    async fn config() -> impl Responder {
        HttpResponse::Ok()
    }
}

#[actix_web::main]
async fn main() {
    let srv = actix_test::start(|| App::new().service(config::config));

    let request = srv.get("/config");
    let response = request.send().await.unwrap();
    assert!(response.status().is_success());
}
