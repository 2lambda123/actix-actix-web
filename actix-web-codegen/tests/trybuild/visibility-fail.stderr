error[E0603]: unit struct `config` is private
  --> tests/trybuild/visibility-fail.rs:15:63
   |
15 |     let srv = actix_test::start(|| App::new().service(config::config));
   |                                                               ^^^^^^ private unit struct
   |
note: the unit struct `config` is defined here
  --> tests/trybuild/visibility-fail.rs:7:5
   |
7  |     #[get("/config")]
   |     ^^^^^^^^^^^^^^^^^
   = note: this error originates in the attribute macro `get` (in Nightly builds, run with -Z macro-backtrace for more info)
